doc/how-to-package-haskell-code.rst

```diff
@@ -0,0 +1,291 @@
+      build-depends: base >=4.11 && <4.12
+      -- hs-source-dirs:
+      default-language: Haskell2010
+
+
+The build information fields listed (but commented out) are just the few
+most important and common fields. There are many others that are covered
+later in this chapter.
+
+Most of the build information fields are the same between libraries and
+executables. The difference is that libraries have a number of "exposed"
+modules that make up the public interface of the library, while
+executables have a file containing a ``Main`` module.
+
+The name of a library always matches the name of the package, so it is
+not specified in the library section. Executables often follow the name
+of the package too, but this is not required and the name is given
+explicitly.
+
+
+Modules included in the package
+-------------------------------
+
+For an executable, ``cabal init`` creates the ``Main.hs`` file which
+contains your program's ``Main`` module. It will also fill in the
+:pkg-field:`executable:main-is` field with the file name of your program's
+``Main`` module, including the ``.hs`` (or ``.lhs``) extension. Other
+modules included in the executable should be listed in the
+:pkg-field:`other-modules` field.
+
+For a library, ``cabal init`` looks in the project directory for files
+that look like Haskell modules and adds all the modules to the
+:pkg-field:`library:exposed-modules` field. For modules that do not form part
+of your package's public interface, you can move those modules to the
+:pkg-field:`other-modules` field. Either way, all modules in the library need
+to be listed.
+
+
+Modules imported from other packages
+------------------------------------
+
+While your library or executable may include a number of modules, it
+almost certainly also imports a number of external modules from the
+standard libraries or other pre-packaged libraries. (These other
+libraries are of course just Cabal packages that contain a library.)
+
+You have to list all of the library packages that your library or
+executable imports modules from. Or to put it another way: you have to
+list all the other packages that your package depends on.
+
+For example, suppose the example ``Proglet`` module imports the module
+``Data.Map``. The ``Data.Map`` module comes from the ``containers``
+package, so we must list it:
+
+::
+
+    library
+      exposed-modules:     Proglet
+      other-modules:
+      build-depends:       containers, base >=4.11 && <4.12
+
+In addition, almost every package also depends on the ``base`` library
+package because it exports the standard ``Prelude`` module plus other
+basic modules like ``Data.List``.
+
+You will notice that we have listed ``base >=4.11 && <4.12``. This gives a
+constraint on the version of the base package that our package will work
+with. The most common kinds of constraints are:
+
+-  ``pkgname >=n``
+-  ``pkgname ^>=n`` (since Cabal 2.0)
+-  ``pkgname >=n && <m``
+-  ``pkgname ==n.*`` (since Cabal 1.6)
```

Comment:
Do we care about the "since" here?

Reply:
Not, sure, but i like to add more explicit examples in a later PR. Let's keep it for now as is.

Reply:
If it's not the reference part of the manual, things like `since Cabal 1.6` are better omitted. Of course, if you feel like leaving it as a future work, that's fine.

Reply:
Removed.